omnigibson/prims/xform_prim.py

```diff
@@ -419,15 +425,15 @@ def _dump_state(self):
         # If we are in a scene, compute the scene-local transform (and save this as the world transform
         # for legacy compatibility)
         if self.scene is not None:
-            pos, ori = T.relative_pose_transform(pos, ori, *self.scene.prim.get_position_orientation())
+            pos, ori = T.mat2pose(self.scene.pose_inv @ T.pose2mat((pos, ori)))
```

Comment:
Let's undo all of this - Frank's new PR will centralize this behavior into the "scene" mode of the pose getters.

Reply:
what behavior will be centralized? the caching mechanism or the computation? This change was done to nontrivially improve the state saving, since every object computes their pose in this manner (so naively we have N calls to the scene.prim.get_position_orientation). 

The refactor is definitely good, but I would prefer to leave this change here so that the caching mechanism is not forgotten about during Frank's follow up PR (not to mention avoiding the inverse transformation in this call too)

Reply:
OK sounds good lets leave it in and merge. We'll unify in his PR